### PR TITLE
Adds UK bank holiday for Queen Elizabeth II’s funeral

### PR DESCRIPTION
**Pull request by freshleafmedia**

### CODE REVIEW

File: src/Yasumi/Provider/UnitedKingdom.php [hunk 1]
```diff
@@ -203,6 +204,33 @@ protected function calculatePlatinumJubileeBankHoliday(): void
         ));
     }
 
+    /**
+     * Queen Elizabeth II’s funeral is an extra bank holiday added on 10 September 2022
```
Comment (sylvaincombes) [inline src/Yasumi/Provider/UnitedKingdom.php:208]: 10 September 2022 > 19 September 2022

Comment (freshleafmedia) [reply src/Yasumi/Provider/UnitedKingdom.php:208]: @sylvaincombes The 10th is the date the bank holiday was announced

Comment (sylvaincombes) [reply src/Yasumi/Provider/UnitedKingdom.php:208]: oh ok sorry I misunderstood and thought it was a typo

Comment (sylvaincombes) [reply src/Yasumi/Provider/UnitedKingdom.php:208]: Thank you for your work, hope it will be merged soon

**Review by stelgenhof — CHANGES_REQUESTED** [review]
@freshleafmedia Thank you very much for this PR! Looks good to me, except it is missing a test assertion for this bank holiday with the United Kingdom provider (see `testBankHolidays`). 

Is this bank holiday observed in the related countries of England, Wales, Scotland and Northern Ireland as well? (I assume they do, but just want to make sure). If so, then the same unit tests need to be added for those countries as well.

**Comment by freshleafmedia** [comment]
@stelgenhof I'm not sure adding it to the `testBankHolidays` method is appropriate because this is a one-off bank holiday for 2022. It is similar to the Platinum Jubilee Bank Holiday. Am I missing something?

It is observed in [England, Wales](https://www.gov.uk/bank-holidays#england-and-wales), [Scotland](https://www.gov.uk/bank-holidays#scotland) and [Northern Ireland](https://www.gov.uk/bank-holidays#northern-ireland).

**Review by stelgenhof — COMMENTED** [review]
@freshleafmedia The unit test for this bank holiday will just assert if it is correct, however the providers have tests that assert the list of bank holidays for a given year. Since this is a new bank holiday, it should be added to that test/assertion.

**Comment by freshleafmedia** [comment]
@stelgenhof I think I understand. Does that mean that the `testBankHolidays` method in `UnitedKingdomTest` should look something like this:

```php
public function testBankHolidays(): void
{
    $this->assertDefinedHolidays([
        'newYearsDay',
        'easterMonday',
        'mayDayBankHoliday',
        'springBankHoliday',
        'secondChristmasDay',
    ], self::REGION, $this->year, Holiday::TYPE_BANK);

    $this->assertDefinedHolidays([
        'queenElizabethFuneralBankHoliday',
    ], self::REGION, 2022, Holiday::TYPE_BANK);
}
```

**Comment by stelgenhof** [comment]
@freshleafmedia Yes, that should do it. Another way would be to build the list of holidays dynamically:

```php
public function testOfficialHolidays(): void                                                                                                                                                                                                                       
{                                                                                                                                                                                                                                                                  
      $holidays = [                                                                                                                                                                                                                                                  
           'newYearsDay',                                                                                                                                                                                                                                             
           'easterMonday',                                                                                                                                                                                                                                            
           'internationalWorkersDay',                                                                                                                                                                                                                                 
           'ascensionDay',                                                                                                                                                                                                                                            
           'pentecostMonday',                                                                                                                                                                                                                                         
           'nationalDay',                                                                                                                                                                                                                                             
           'assumptionOfMary',                                                                                                                                                                                                                                        
           'allSaintsDay',                                                                                                                                                                                                                                            
           'christmasDay',                                                                                                                                                                                                                                            
           'secondChristmasDay',                                                                                                                                                                                                                                      
       ];                                                                                                                                                                                                                                                             
                                                                                                                                                                                                                                                                      
        $year = $this->generateRandomYear();                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                       
       if ($year >= Luxembourg::EUROPE_DAY_START_YEAR) {                                                                                                                                                                                                              
            $holidays[] = 'europeDay';                                                                                                                                                                                                                                 
       }                                                                                                                                                                                                                                                              
                                                                                                                                                                                                                                                                       
       $this->assertDefinedHolidays($holidays, self::REGION, $year, Holiday::TYPE_OFFICIAL);      
}
 ```
(Taken from the Luxembourg provider test).

**Comment by kissifrot** [comment]
Hello, would be cool if this PR was merged/released as it would be a bit late otherwise :stuck_out_tongue:

**Review by stelgenhof — APPROVED** [review]

**Comment by stelgenhof** [comment]
Approved the PR as the bank holiday is occurring today. Will add the remaining unit test afterwards.

**Pull request merged**